rotkehlchen/chain/evm/decoding/weth/decoder.py

```diff
@@ -98,6 +106,16 @@ def _decode_deposit_event(self, context: DecoderContext) -> DecodingOutput:
         return DecodingOutput(event=in_event)
 
     def _decode_withdrawal_event(self, context: DecoderContext) -> DecodingOutput:
+        # This only for withdraw tx on Scroll, Because the unwrap method on WETH on scroll
+        # generates an additional log (Transfer) which not exists on other chains.
+        # So because of this additional log, this method will be called twice instead of one,
+        # therefore we need to skip the second call to avoid creating extra events.
+        # Check this for example:
+        # A withdraw on Scroll: https://scrollscan.com/tx/0x88f49633073a7667f93eb888ec2151c26f449cc10afca565a15f8df68ee20f82#eventlog
+        # A withdraw on Ethereum: https://etherscan.io/tx/0xe5d6fa9c60f595a624e1dd71cb3b28ba9260e90cf3097c5b285b901a97b6246d#eventlog
+        if len(context.decoded_events) == 3:
+            return DecodingOutput()
```

Comment:
```suggestion
        # This is only for withdrawals on Scroll because the unwrap method on WETH on scroll
        # generates an additional log (Transfer) which doesn't exists on other chains.
        # Because of this additional log this method will be called twice instead of once,
        # therefore we need to skip the second call to avoid creating extra events.
        # Check this for example:
        # A withdraw on Scroll: https://scrollscan.com/tx/0x88f49633073a7667f93eb888ec2151c26f449cc10afca565a15f8df68ee20f82#eventlog
        # A withdraw on Ethereum: https://etherscan.io/tx/0xe5d6fa9c60f595a624e1dd71cb3b28ba9260e90cf3097c5b285b901a97b6246d#eventlog
        if self.evm_inquirer.chain_id == ChainID.SCROLL and len(context.decoded_events) == 3:
            return DEFAULT_DECODING_OUTPUT
```

but this is a bad check because if another contracts calls the wrapper I believe this can fail. The case being a contract that unstakes and unwraps for example. This I believe is not possible for normal users but safes can trigger it if I'm not mistaken. I would move this check after you read the `withdrawn_amount`and do something like

```python
if self.evm_inquirer.chain_id == ChainID.SCROLL:
    for event in context.decoded_events:
        if event.counterparty == CPT_WETH and event.asset == A_WETH_SCROLL and event.amount == withdrawn_amount:
             return DEFAULT_DECODING_OUTPUT  # we have already decoded this event
```